x/stakeibc/keeper/lsm_test.go

```diff
@@ -169,3 +169,42 @@ func (s *KeeperTestSuite) TestShouldQueryValidatorExchangeRate() {
 		s.Require().Equal(tc.expectedShouldQuery, actualShouldQuery, tc.name)
 	}
 }
+
+func (s *KeeperTestSuite) TestRefundLSMToken() {
+	liquidStakerAddress := s.TestAccs[0]
+	hostZoneAddress := types.NewZoneAddress(HostChainId)
+
+	// Fund the module account with the LSM token
+	lsmTokenIBCDenom := "ibc/cosmosvalXXX"
+	stakeAmount := sdk.NewInt(1000)
+	lsmToken := sdk.NewCoin(lsmTokenIBCDenom, stakeAmount)
+	s.FundAccount(hostZoneAddress, lsmToken)
+
+	// Setup the liquid stake object that provides the context for the refund
+	liquidStake := types.LSMLiquidStake{
+		Staker: liquidStakerAddress,
+		HostZone: types.HostZone{
+			Address: hostZoneAddress.String(),
+		},
+		LSMIBCToken: lsmToken,
+	}
+
+	// Refund the token and check that it has been successfully transferred
+	err := s.App.StakeibcKeeper.RefundLSMToken(s.Ctx, liquidStake)
+	s.Require().NoError(err, "no error expected when refunding LSM token")
+
+	stakerBalance := s.App.BankKeeper.GetBalance(s.Ctx, liquidStakerAddress, lsmTokenIBCDenom)
+	s.CompareCoins(lsmToken, stakerBalance, "staker should have received their LSM token back")
+
+	moduleBalance := s.App.BankKeeper.GetBalance(s.Ctx, hostZoneAddress, lsmTokenIBCDenom)
+	s.True(moduleBalance.IsZero(), "module account should no longer have the LSM token")
+
+	// Attempt to refund again, it should fail from an insufficient balance
+	err = s.App.StakeibcKeeper.RefundLSMToken(s.Ctx, liquidStake)
+	s.Require().ErrorContains(err, "insufficient funds")
```

Comment:
Should we do this with different error types instead of string checking?

Reply:
Good question! I've found using the string approach is a little easier to manage cause once you wrap an error, you can no longer check the type

For instance, if you return 
```go
errorsmod.Wrapf(types.SomeErrorType, "some additional context")
```
then you _cannot_ do the following
```go
s.Require().ErrorIs(types.SomeErrorType, err) 
```

And we wrap most of our errors. Also, there's often a whole stack trace for the error, so it's nice to just check for the part that relevant to the test case